File: backend-dotnet/SahayaSetu/Services/IDonorService.cs
using SahayaSetu.DTO;

namespace SahayaSetu.Services
{
    public interface IDonorService
    {
        Task<List<RequestResponseDto>> GetAllOpenRequestsAsync();
        Task<List<RequestResponseDto>> GetAllOpenRequestsForDonorAsync(int donorId);
        Task<List<FundraiserRequestResponseDto>> GetOpenFundraisersAsync();
        Task DonateToFundraiserAsync(int donorId, int requestId, DonationDto dto);
        Task ApplyForVolunteerAsync(int donorId, int requestId);
        Task FulfillResourceRequestAsync(int donorId, int requestId, int quantity);
        Task<ContributionSummaryDto> GetDonorContributionsAsync(int donorId);
    }
}

File: backend-java/sahayaSetu/src/main/java/com/sahayaSetu/services/IDonorService.java
package com.sahayaSetu.services;

import com.sahayaSetu.dtos.*;

import java.util.List;

public interface IDonorService {
    List<RequestResponseDto> getAllOpenRequests();
    List<RequestResponseDto> getAllOpenRequestsForDonor(Long donorId);
    List<FundraiserRequestResponseDto> getOpenFundraisers();
    void donateToFundraiser(Long donorId, Long requestId, DonationDto dto);
    void applyForVolunteer(Long donorId, Long requestId);
    void fulfillResourceRequest(Long donorId, Long requestId, int quantity);
    ContributionSummaryDto getDonorContributions(Long donorId);
}
